monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/MapParallelOrderedObservable.scala

```diff
@@ -105,7 +105,7 @@ private[reactive] final class MapParallelOrderedObservable[A, B](
                     composite -= head.cancelable
                   case Failure(ex) =>
                     lastAck = Stop
-                    composite -= head.cancelable
+                    composite.cancel() // cancel the whole downstream on error
```

Comment:
I think it will be better to `composite.cancel()` in `process` method so as soon as the task is completed with failure

Reply:
@TapanVaishnav Sorry for nitpicking but this line should be redundant now :D

Reply:
Ahhh, because `process` calls `sendDownstreamOrdered` which then calls `onNext` and which then calls the `process` again. And if we stop the stream in the `process` itself then need not worry about `sendDownstreamOrdered` method.

Reply:
@Avasil I have removed the above line and believe that we needn't to add `composite -= head.cancelable` again, right?

Reply:
Yes, I think there are some redundant operators there but it's OK, I have to sit one day and refactor entire operator, it's a bit of a mess :D